### ISSUE TITLE
port for arduino

### ISSUE THREAD

**Issue by ladyada** [ghent360/RTD-2660-Programmer#3]
hiya, i did a port for arduino - i had been using the $75 standalone programmers (they can store the BIN in a flash chip for untethered programming but requires win XP). using an arduino compat+SD card is way superior!

https://github.com/adafruit/Adafruit_RTD266X_I2CFlasher

works great, and am compiling some custom RTD firmware, with success so far. thanks for the i2c code :) life is :100:

**Comment by ghent360** [comment]
Cool, now if we can master an open source firmware for the rtd2660, that
would be great.

**Comment by ladyada** [comment]
ahhh - im not sure i want to rewrite the RTD code - maybe better to just release diff's from the "RTD2660_AV1_AV2_081015" package? the base works, that way we just tweak it as needed?

**Comment by ghent360** [comment]
It can probably be compiled in a way that the LCD timing table can be
located in the binary and have a tool that patches the timings in the
compiled code.

What RDT2660 board are u using?

**Comment by ladyada** [comment]
handmade board, but its electrically similar to PCB800116 (HDMI -> TTL w/RTD2662)
so far just got basic compiling + managed to invert Hsync and Vsync. 
pretty much just aiming to get 480x800 out (not 800x480). currently using https://github.com/avtehnik/RTD2662 as a base. binary patching of Panel2's data will probably be fairly easy once that is working!

eventually would like to pair with SSD2828 for DSI/MIPI - most small MIPI displays are portrait

**Comment by ghent360** [comment]
Are you planning to rotate the input? I don't think the rtd2660 it is
capable.

**Comment by ladyada** [comment]
yah i dont think its possible to rotate but that's easy to do in the OS - at least Win & Linux can switch a monitor from portrait to landscape so you'd have to do that as well. not ideal but not a lot of other options!

**Comment by ghent360** [comment]
I looked at a few MIPI displays and most of them can do hardware 90 deg rotation. Thanks for the SSD2828 tip. I was looking for a chip like that for quite some time.

**Comment by ladyada** [comment]
cool - i haven't seen that yet but maybe i'm not looking in the right place. is the rotation pin-strapped or is there a DSI command? i know there's video & cmd mode but have not done any actual work with MIPI yet

**Comment by ghent360** [comment]
It's a DSi command.

**Comment by goudon007** [comment]
Hi guys,
Like you idea about connecting this RGB interface with a MIPI interface.
I have already implemented some part of this. MY mipi interface is already working.
But, RTD2660 some have is not giving me the right Pclk, even though I managaed to edit the EDID information in the eprom, which is used to program the RTD2660. I mean, the datasheet for RTD2660 has a whole lot of info on programming it, but its so breifly written and not elaborative to edit. Can somebody help me ?
